harness/determined/estimator/_estimator_trial.py

```diff
@@ -641,13 +644,14 @@ def compute_validation_metrics(self) -> workload.Response:
         return {"validation_metrics": metrics}
 
     def average_metrics(self, metrics: Dict[str, Any]) -> Optional[Dict[str, Any]]:
-        # The chief receives the metric from every worker and computes
-        # the average.
+        # The chief receives the metric from every worker and computes the average.
         check.true(self.hvd_config.use)
         if self.is_chief:
-            self.train_process_comm_chief = cast(ipc.ZMQServer, self.train_process_comm_chief)
-            logging.debug(f"Chief {hvd.rank()} beginning receiving validation metrics.")
-            worker_metrics = self.train_process_comm_chief.barrier(num_connections=hvd.size() - 1)
+            self.train_process_comm_chief = cast(
+                ipc.ZMQBroadcastServer, self.train_process_comm_chief
+            )
+            logging.debug("Chief {hvd.rank()} beginning receiving validation metrics.")
+            worker_metrics, _ = self.train_process_comm_chief.gather_with_polling(lambda: None)
```

Comment:
non-blocking: would be nice to have this re-use the `allgather_metrics` function, it would introduce unnecessary overhead of an extra broadcast.

Reply:
we talked offline, and we are just leaving this as-is